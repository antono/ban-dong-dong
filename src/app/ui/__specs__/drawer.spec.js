import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Drawer from '../drawer';

function setup(propsExtension = {}) {
  const props = Object.assign({
    isOpen: false,
    handleOpenLink: jest.fn(),
    handleCloseDrawer: jest.fn(),
  }, propsExtension);

  const component = shallow(<Drawer.WrappedComponent {...props} />);

  return { props, component };
}

describe('Drawer', () => {
  it('matches snapshot', () => {
    const { component } = setup();
    expect(toJSON(component)).toMatchSnapshot();
  });
});
