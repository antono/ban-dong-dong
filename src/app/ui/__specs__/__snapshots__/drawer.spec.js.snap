exports[`Drawer matches snapshot 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={false}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}>
  <MenuItem
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}>
    Welcome
  </MenuItem>
  <MenuItem
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}>
    Sign In
  </MenuItem>
</Drawer>
`;
